### PR TITLE
Mark archived Editions and GuideEditions as not previewable

### PR DESCRIPTION
**Pull request by issyl0**
- For https://github.com/alphagov/publisher/pull/590.

### CODE REVIEW

File: app/models/edition.rb [hunk 1]
```diff
@@ -266,7 +266,7 @@ def has_video?
   end
 
   def safe_to_preview?
-    true
+    true unless archived?
```
Comment (cbaines) [inline app/models/edition.rb]: This will return nil, rather than false for archived editions. It might be nicer to write it as `not archived?`?

Comment (issyl0) [inline app/models/edition.rb]: Fixed with `!archived?`.

File: app/models/guide_edition.rb [hunk 2]
```diff
@@ -14,6 +14,6 @@ def has_video?
   end
 
   def safe_to_preview?
-    parts.any? and parts.first.slug.present?
+    parts.any? && parts.first.slug.present? && !edition.archived?
```
Comment (cbaines) [inline app/models/guide_edition.rb]: You might want to consider changing this to use safe_to_preview? from the superclass?

File: app/models/edition.rb (hunk 1 above)
Comment (cbaines) [inline app/models/edition.rb]: Works for me :+1:

File: app/models/guide_edition.rb (hunk 2 above)
Comment (whoojemaflip) [inline app/models/guide_edition.rb]: agree, I think you could replace `!edition.archived?` here with a call to `super`

Comment (issyl0) [inline app/models/guide_edition.rb]: I changed this to be `super && [guide stuff]`, so it gets `!archived?` from Edition.

Comment (cbaines) [inline app/models/guide_edition.rb]: :+1:

**Pull request merged**